Assets/Scripts/Core/GridCombatSystem.cs

```diff
@@ -193,5 +195,41 @@ private enum State {
 		public CoreUnit GetActiveUnit() {
 			return unitGridCombat_;
 		}
+
+		// It is possible to add another float variable WeaponModifierHitChance to this method, then adjust float hitChance accordingly.
+		private float RangedHitChance(Vector3 player, Vector3 target, float weaponMaxRange) {
+			Grid<Tilemap.Node> grid = GameController.Instance.GetGrid();
+			List<Vector3> points = Line(player,target);
+			float hitChance = 100f;
+
+			if(weaponMaxRange >= Vector3.Distance(player, target)) {
+				// Target is in range
+				foreach(Vector3 v in points) {
+					// Calculates hitchance
+					hitChance -= grid.GetGridObject(v).hitChanceModifier;
+				}
+			}
+			return hitChance;
+		}
+
+		// The methods `Line()`, `Lerp()` and `DiagonalDistance()` are from https://www.redblobgames.com/grids/line-drawing.html and adjusted accordingly.
+
+		// Calculates the line between two Vector3's and returns N nodes between them.
+		private List<Vector3> Line(Vector3 p0, Vector3 p1) {
+			List<Vector3> points = new List<Vector3>();
+			float N = DiagonalDistance(p0, p1);
+			for (int step = 0; step <= N; step++) {
+				float t = N == 0? 0.0f : step / N;

```

Comment:
`N` isn't a descriptive name, this can and should be a more descriptive name. Also missing a space, `0? 0.0f` should be `0 ? 0.0f`.